### PR TITLE
Add jupyter tutorial

### PR DESCRIPTION
**Pull request by ATriantafyllopoulos**
### Summary

This adds the much anticipated jupyter notebook tutorial (#64). It also adds the `EmoDB` data set to have a minimal example.

### Proposed Changes

* Adds `EmoDB` data set
* Adds introduction jupyter notebook tutorial

### Discussion
In order to make the notebook executable by everyone, I added two emodb files to the package. I think this is better than doing the example simply with white noise.

### CODE REVIEW

File: audtorch/datasets/emodb.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class EmoDB(AudioDataset):
+    r"""EmoDB data set.
+
+    Open and publicly available data set of acted emotions:
+    http://www.emodb.bilderbar.info/navi.html
+
+    EmoDB is a small audio data set collected in an anechoic chamber in the
+    Technical University of Berlin, it contains 5 male and 5 female speakers,
```
Comment (hagenw) [inline audtorch/datasets/emodb.py]: It's a little bit strange, but the official name of the University is `Technische Universität Berlin` or `Berlin Institute of Technology`. So please pick one of those.

File: docs/tutorials/introduction.ipynb [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+   "source": [
+    "## Preliminaries\n",
+    "\n",
+    "* [PyTorch](https://pytorch.org/) already has an inteface for data sets, aptly called [Dataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset) \n",
+    "* It then wraps this interface with a [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) that efficiently allows us to loop through the data in parallel, and takes care of the random order as well\n",
+    "* All we need to do is implement the [Dataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset) interface to get the input for the model and the labels\n",
+    "* **However**, it is not easy for beginners to see how one can go from a bunch of files in their hard drive, to the features that will be used as input in a machine learning model\n",
+    "* **Thankfully**, `audtorch` is there to take of all that for you :-)\n",
+    "\n",
+    "Before you start, you might want to familiarize yourselves with [PyTorch's data pipeline] (https://pytorch.org/docs/stable/data.html)"
```
Comment (hagenw) [inline docs/tutorials/introduction.ipynb]: There is a typo, should be `[PyTorch's data pipeline](https://pytorch.org/docs/stable/data.html)`

File: docs/tutorials/introduction.ipynb [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import audtorch\n",
+    "import IPython.display as ipd"
```
Comment (hagenw) [inline docs/tutorials/introduction.ipynb]: This fails as `matplotlib` is not in `docs/requirements.txt`

File: docs/index.rst [hunk 4]
```diff
@@ -29,3 +29,9 @@
     api-utils
     refs
     genindex
+
+.. toctree::
+    :caption: Tutorials
+    :hidden:
+
+    tutorials/introduction
```
Comment (hagenw) [inline docs/index.rst]: I would move this section before the API documentation

**Comment by hagenw** [comment]
Cool contribution, thank you very much.

Please also rebase the master in order to get the failing tests passing.

**Comment by ATriantafyllopoulos** [comment]
I was now able to solve the automatic execution, and all the review comments have been addressed as well.

**Pull request merged**